extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -298,120 +298,108 @@ void ConsumeWindowsEventLog::onSchedule(const std::shared_ptr<core::ProcessConte
   logger_->log_trace("Successfully configured CWEL");
 }
 
+bool ConsumeWindowsEventLog::commitAndSaveBookmark(const std::wstring &bookmark_xml, const std::shared_ptr<core::ProcessSession> &session) {
+  {
+    const TimeDiff time_diff;
+    session->commit();
+    logger_->log_debug("processQueue commit took %" PRId64 " ms", time_diff());
+  }
+
+  if (!bookmark_->saveBookmarkXml(bookmark_xml)) {
+    logger_->log_error("Failed to save bookmark xml");
+  }
+
+  if (session->outgoingConnectionsFull("success")) {
+    logger_->log_debug("Outgoing success connection is full");
+    return false;
+  }
+
+  return true;
+}
+
+std::wstring ConsumeWindowsEventLog::processEventLogs(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session,
+    size_t& processed_event_count, const EVT_HANDLE& event_query_results) {
```

Comment:
I would do this the other way round: return the number of events processed, and use an out parameter for the bookmark XML.  Not important, but I think that is a more common pattern.

Another option would be to return a struct `{ size_t processed_event_count; std::wstring bookmark_xml; }`.

Reply:
As both are output parameters I decided to return a tuple with both values. Added in [24855ee](https://github.com/apache/nifi-minifi-cpp/pull/891/commits/24855eeb70fffe2f33b065bcd75dc77f8d5aa3f5)